src/upchunk.ts

```diff
@@ -290,20 +390,46 @@ export class UpChunk {
     this.success = false;
     this.nextChunkRangeStart = 0;
 
+    if (options.useFileSliceFallback) {
+      const readableStreamErrorCallback = (event: CustomEvent) => {
+        // In this case, assume the error is a result of file reading via ReadableStream.
+        // Retry using ChunkedFileIterable, which reads the file into memory instead
+        // of a stream.
+        if (this.chunkedIterable.error) {
+          console.warn(
+            `Unable to read file of size ${this.file.size} bytes via a ReadableStream. Falling back to in-memory FileReader!`
+          );
+          event.stopImmediatePropagation();
+
+          // Re-set everything up with the fallback iterable and corresponding
+          // iterator
+          this.chunkedIterable = new ChunkedFileIterable(this.file, {
+            ...options,
+            defaultChunkSize: options.chunkSize,
+          });
+          this.chunkedIterator = this.chunkedIterable[Symbol.asyncIterator]();
+          this.getEndpoint().then(() => {
+            this.sendChunks();
+          });
+          this.off('error', readableStreamErrorCallback);
+        }
+      };
+      this.on('error', readableStreamErrorCallback);
+    }
+
     // Types appear to be getting confused in env setup, using the overloaded NodeJS Blob definition, which uses NodeJS.ReadableStream instead
     // of the DOM type definitions. For definitions, See consumers.d.ts vs. lib.dom.d.ts. (CJP)
-    this.chunkedStreamIterable = new ChunkedStreamIterable(
+    this.chunkedIterable = new ChunkedStreamIterable(
```

Comment:
went ahead and renamed these guys to be more generic, given the potential fallback.